### ISSUE TITLE
[com_fields] Fields and groups in languages not working

### ISSUE THREAD

**Issue by JUG-FN** [joomla/joomla-cms#14865]
### Steps to reproduce the issue

Joomla 3.7 Custome Fields for User. 
I have a test page 3.7 Beta 3 with three languages created. New groups and fields for EACH language created for user registration. It does not seem to work. All groups (and fields) of languages are ALWAYS displayed!

Groups and fields with the corresponding languages are always displayed in ALL languages. So EN also in DE...

You can see it here:
http://joomla03.jug-friedrichshafen.de/de/component/users/?view=registration&Itemid=103

### System information (as much as possible)
Linux info 3.0 #1337 SMP Tue Jan 01 00:00:00 CEST 2000 all GNU/Linux
Datenbankversion 	5.5.54-0+deb7u2-log
Datenbankzeichensatz 	utf8_general_ci
Datenbankverbindungszeichensatz 	utf8mb4_general_ci
PHP-Version 	7.0.16
Webserver 	Apache
PHP-Interface für den Webserver 	cgi-fcgi
Joomla!-Version 	Joomla! 3.7.0-beta3 Beta [ Amani ] 21-February-2017 15:03 GMT
Joomla!-Plattform-Version 	Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
Browsererkennung 	Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0

**Comment by brianteeman** [comment]
I can confirm this error

**Comment by Bakual** [comment]
From memory, I think the language filter relates to the item language, not the site language. But in the registration form we don't really have an item. So that may be where this issue comes from.

**Comment by infograf768** [comment]
It does work fine for articles (for custom fields at least) on a multilanguage site.

**Comment by Bakual** [comment]
Yep, because articles have a language property, the check works fine there. But since the registration form and likely also the contact form doesn't have a language property, it will show all fields.
The same happens by the way when you create a new article where the language isn't set yet, there all fields will be shown as well.

For the registration and contact form, the expected behavior is clear to only show the fields matching the site language. But what would you expect for new items (eg new article)?
Technically it's the same code that is run there (onContentPrepareForm event).

**Comment by brianteeman** [comment]
Wouldn't the available fields change when you change language in the same way the available fields change when you change category?

**Comment by Bakual** [comment]
Apparently not. Only after saving the article with a language set, the fields are filtered. Probably an issue by itself.

**Comment by laoneo** [comment]
A solution for the initial issue description can be to fall back to the language of the site, when the item doesn't have a language property.
The issue discovered by @Bakual and @brianteeman should probably be solved the same way as with categories, with a page reload when the language changes and some intelligence when a new article is created.

**Comment by infograf768** [comment]
> A solution for the initial issue description can be to fall back to the language of the site, when the item doesn't have a language property.

Not sure about this one. I explain:
On a monolanguage site where multiple languages are installed, a user may look at the site in a specific language (depending on her/his choice) **for the UI** but the contents are the same.

**Comment by Bakual** [comment]
@infograf768 On a site with only one language, the user doesn't have fields assigned to specific languages (since there are no other content languages), so it's no issue at all.

**Comment by laoneo** [comment]
As long as the item (article) has a language property, then nothing changes, but as soon as the item has no language property, then the language from the site is used. That was my intention.

**Comment by infograf768** [comment]
@Bakual
> On a site with only one language, the user doesn't have fields assigned to specific languages (since there are no other content languages), so it's no issue at all.

Not true, now, when someone install a language, (and although I was against it), a content language is always created. Even if it is not published, it will display in the Language field.
Therefore someone may assign a language to a field or fieldgroup on a monolanguage site.
Let's not forget that multiple languages can be installed on a monolanguage site and they are not used the same way
 as on a multilingual site.

**Comment by Bakual** [comment]
> Therefore someone may assign a language to a field or fieldgroup on a monolanguage site.

Yes, of coursse the user can misconfigure a field, but honestly then I would also expect the field to be filtered by the language even on a site that is not set up as multilingual.

**Comment by laoneo** [comment]
Please test pr #15223 which should solve the issue.

**Comment by JUG-FN** [comment]
I have tested in RC2 and now it works fine. thank you.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14865">issues.joomla.org/tracker/joomla-cms/14865</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
closed as Issue solved.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14865">issues.joomla.org/tracker/joomla-cms/14865</a>.</sub>

**Comment by ynzejan** [comment]
What is meant by 'the issue is closed'?
I have updated to Joomla 3.8.0 and I am experiencing the problem as raised.
I have a multilingual (main language NL and second language GB) site which was setup without data.
I have defined two user groups (one for each language) in the Users menu and defined a number of fields for each group (each field assigned to the appropriate language).
All fields are required.
In the backend when I define a new user I am forced to enter all fields for both field groups (although I have defined one main language).
In the frontend when showing a user profile both field groups are shown.

So, as far as I am concerned the issue is far from resolved.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14865">issues.joomla.org/tracker/joomla-cms/14865</a>.</sub>

**Comment by ynzejan** [comment]
I have to correct myself. The initial setup for my website was with sample data and multilanguage support was added later.
So, does that mean that I cannot resolve the issue in the current setup and have to setup a fresh install without sample data and including multilanguage support upfront? And have the hassle of in one way or the other migrate everything I have setup so far from the current to that new website setup (which would be quite a shame)?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14865">issues.joomla.org/tracker/joomla-cms/14865</a>.</sub>

**Comment by ghost** [comment]
@ynzejan can you please open an new Issue as comment on closed issues didn't get much Attention, thanks.
